scripts/corelli/calibration/database.py

```diff
@@ -7,118 +7,198 @@
 
 # File that defines functionality for Corelli calibration database
 
-import pathlib
 from datetime import datetime
+import enum
+import pathlib
 import re
-from typing import Optional
+from typing import List, Optional, Tuple, Union
 
-from mantid.dataobjects import TableWorkspace
-from mantid.api import WorkspaceGroup, Workspace
+from mantid.dataobjects import EventWorkspace, TableWorkspace,  Workspace2D
+from mantid.api import mtd, Workspace, WorkspaceGroup
 from mantid.simpleapi import CreateEmptyTableWorkspace, SaveNexusProcessed, LoadNexusProcessed
 
+# Custom type aliases
+CalibrationInputSetTypes = Union[str, TableWorkspace, Workspace2D, List[TableWorkspace], List[str], WorkspaceGroup]
+InputWorkspaceTypes = Union[str, Workspace2D, EventWorkspace]
+
+
+class TableType(enum.Enum):
```

Comment:
Bikeshedding, not sure if we need it as a class. I'm fine either way, I just kept it procedural for consistency.

Reply:
It might not be necessary, you're right, but I'd like to keep if for now it that's OK by you

Reply:
That's fine.